### PR TITLE
Merge Jiarong's branch

### LINKED ISSUE #5: KK: "Save Preset" button missing following refactor
The "Save Preset" button seems to have been removed from KK Stiletto following a GUI refactor. It's not clear to me how saving is supposed to work now. I don't see any functions that call `SaveHeelFile` in `master`.

If this is unintentional and is just a result of a half-finished refactor, I might be up for contributing a PR to fix this. But I figured I should ask first, to see if I've missed something.

### PR DESCRIPTION
**Pull request by Splendide-Imaginarius**
Fixes https://github.com/IllusionMods/Stiletto/issues/5 . Also fixes a bug that broke the knee bend feature.

I've tested the branch for a few days, and it seems to work OK. Here are the caveats:

* While I don't see any obvious bugs in the behavior, I don't have the necessary knowledge to properly review all of the code changes themselves.
* I didn't test Studio mode at all, so if there are bugs there, I wouldn't have noticed.
* There's a lot of code churn here, which might make https://github.com/IllusionMods/Stiletto/issues/3 more difficult to solve.

Personally I don't think any of these caveats should block a merge, but it's not my repo, so my opinion matters little.

### CODE REVIEW

**Comment by Splendide-Imaginarius** [comment]
> There's a lot of code churn here, which might make https://github.com/IllusionMods/Stiletto/issues/3 more difficult to solve.

There is a chance that I might try to unify the KK and KKS codebases sometime after this PR is merged, but this may or may not happen soon, and my C# skills may or may not be enough to do it successfully.

File: Stiletto/Stiletto.cs [hunk 1]
```diff
@@ -15,195 +11,171 @@ namespace Stiletto
     [BepInPlugin(GUID, nameof(Stiletto), Version)]
     public class Stiletto : BaseUnityPlugin
     {
-        public const string GUID = "com.essu.stiletto";
-        public const string Version = "1.4.1";
+        public const string GUID = "com.essu.stiletto.custom";
```
Comment (ManlyMarco) [inline Stiletto/Stiletto.cs]:
```suggestion
        public const string GUID = "com.essu.stiletto";
```

**Review by ManlyMarco — COMMENTED** [review]
This looks like an almost completely different code base than what's currently in the repo. While it improves in a lot of areas it also regresses in others (e.g. no longer adding user-friendly maker controls, vastly increased code complexity, not using some of the newer features, etc.), so I'm hesitant to just blindly merge it.

To merge this with the KKS version it seems like the best way would be to scrap the current KKS codebase and port this version, adding any KKS specific hooks/tweaks that were added to the current version. You can find out what was changed in the KKS version in #2.

**Comment by ManlyMarco** [comment]
Since this repository doesn't have active work happening on it and doesn't host releases so far, I guess there's no real harm in merging this and improving it from there. If you could handle the KKS port it would be great.

I added a legacy v1 branch in case the old version needs to be updated.

**Comment by Splendide-Imaginarius** [comment]
> While it improves in a lot of areas it also regresses in others (e.g. no longer adding user-friendly maker controls, vastly increased code complexity, not using some of the newer features, etc.)

Could you clarify which maker controls and newer features you noticed are missing? I'd be happy to try my hand at re-adding them, if it's within my skill level.

**Comment by ManlyMarco** [comment]
> Could you clarify which maker controls and newer features you noticed are missing? I'd be happy to try my hand at re-adding them, if it's within my skill level.

My bad, looks like the maker controls did get kept, just moved to a different file. In that case it's just about not using latest BepInEx and other nuget packages, which results in unnecessary reflection and other minor inefficiencies. Overall it's not a big issue, all it really does is make the codebase harder to understand.

**Pull request merged**